sui_programmability/examples/games/sources/vdf_based_lottery.move

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// A basic lottery game that depends on user-provided randomness which is processed by a verifiable delay function (VDF)
+/// to make sure that it is unbiasable. 
+/// 
+/// During the submission phase, players can buy tickets. When buying a ticket, a user must provide some randomness `r`. This 
+/// randomness is added to the combined randomness of the lottery, `h`, as `h = Sha2_256(h, r)`.
+/// 
+/// After the submission phase has ended, the combined randomness is used to generate an input to the VDF. Anyone may now
+/// compute the output and submit it along with a proof of correctness to the `complete` function. If the output and proof 
+/// verifies, the game ends, and the hash of the output is used to pick a winner.
+/// 
+/// The outcome is guaranteed to be fair if:
+///  1) At least one player contributes true randomness,
+///  2) The number of iterations is defined such that it takes at least `submission_phase_length` to compute the VDF.
+module games::vdf_based_lottery {
+    use games::drand_lib::safe_selection;
+    use sui::clock::Clock;
+    use std::hash::sha2_256;
+    use sui::vdf::{hash_to_input, vdf_verify};
+
+    /// Error codes
+    const EGameNotInProgress: u64 = 0;
+    const EGameAlreadyCompleted: u64 = 1;
+    const EInvalidTicket: u64 = 3;
+    const ESubmissionPhaseInProgress: u64 = 4;
+    const EInvalidVdfProof: u64 = 5;
+    const ESubmissionPhaseFinished: u64 = 6;
+
+    /// Game status
+    const IN_PROGRESS: u8 = 0;
+    const COMPLETED: u8 = 2;
```

Comment:
ooc in the erros above we don't have # 2, and here we don't have # 1, is this on purpose?

Reply:
Yea, that's a bit confusing. Fixing in next commit.